scrapy/downloadermiddlewares/httpcompression.py

```diff
@@ -18,11 +18,14 @@
 class HttpCompressionMiddleware:
     """This middleware allows compressed (gzip, deflate) traffic to be
     sent/received from web sites"""
+    def __init__(self, stats=None):
+        self.stats = stats
+
     @classmethod
     def from_crawler(cls, crawler):
         if not crawler.settings.getbool('COMPRESSION_ENABLED'):
             raise NotConfigured
-        return cls()
+        return cls(crawler.stats)
```

Comment:
This is the other change I mentioned, in case of existing subclasses without `from_crawler` overwritten but with an `__init__` method that takes no parameters:

```suggestion
        try:
            return cls(stats=crawler.stats)
        except TypeError:
            result = cls()
            result.stats = crawler.stats
            return result
```